lambda-functions/database-factory-horizontal-scaling/main.go

```diff
@@ -0,0 +1,331 @@
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get initiate AWS session")
+	}
+	client := resourcegroupstaggingapi.New(sess)
+
+	for _, vpc := range vpcList {
+		log.Infof("Checking DB clusters for VPC (%s)", vpc)
+		dbClusters, err := client.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+			TagFilters: []*resourcegroupstaggingapi.TagFilter{
+				{
+					Key:    aws.String("DatabaseType"),
+					Values: []*string{aws.String(os.Getenv("RDSMultitenantDBClusterTagDatabaseType"))},
+				},
+				{
+					Key:    aws.String("VpcID"),
+					Values: []*string{aws.String(vpc)},
+				},
+				{
+					Key:    aws.String("Purpose"),
+					Values: []*string{aws.String(os.Getenv("RDSMultitenantDBClusterTagPurpose"))},
+				},
+			},
+			ResourceTypeFilters: []*string{aws.String("rds:cluster")},
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get available multitenant DB clusters")
+		}
+		if len(dbClusters.ResourceTagMappingList) > 0 {
+			bool, err := checkVPCScaling(dbClusters, vpc)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to check if cluster scaling is needed")
+			}
+			if bool {
+				log.Infof("Initiating new RDS cluster deployment in vpc (%s)", vpc)
+				err = requestDeployCluster(vpc)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to request a new RDS cluster deployment")
+				}
+			}
+		}
+
+	}
+
+	return nil, nil
+}
+
+func requestDeployCluster(vpc string) error {
+	applyBool, err := strconv.ParseBool(os.Getenv("TerraformApply"))
+	if err != nil {
+		return errors.Wrap(err, "failed to parse bool from TerraformApply string")
+	}
+
+	requestBody, err := json.Marshal(map[string]interface{}{
+		"vpcID":                 vpc,
+		"environment":           os.Getenv("Environment"),
+		"stateStore":            os.Getenv("StateStore"),
+		"apply":                 applyBool,
+		"instanceType":          os.Getenv("DBInstanceType"),
+		"backupRetentionPeriod": os.Getenv("BackupRetentionPeriod"),
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal request body")
+	}
+
+	resp, err := http.Post(os.Getenv("DatabaseFactoryEndpoint"), "application/json", bytes.NewBuffer(requestBody))
+	if err != nil {
+		return errors.Wrap(err, "failed to POST request")
+	}
+
+	defer resp.Body.Close()
+	statusOK := resp.StatusCode >= 200 && resp.StatusCode < 300
+	if !statusOK {
+		return errors.Errorf("Non-OK HTTP status:%v", resp.StatusCode)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed tο read response body")
+	}
+	var databaseFactoryRequest DatabaseFactoryRequest
+	err = json.Unmarshal(body, &databaseFactoryRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed tο unmarshal database factory request")
+	}
+	err = sendMattermostNotification(databaseFactoryRequest, "A new RDS Cluster was successfully requested in the Database Factory.")
+	if err != nil {
+		return errors.Wrap(err, "failed tο send Mattermost notification")
```

Comment:
Should we return an error here or just log the error? I am honestly not sure what would be best so 0/5.

Reply:
I am 0/5 on that as well. I think we can keep it for now because if we create a cluster we want to know about it and if that fails its sth we need to look at.